### DISCUSSION TITLE
Need help in the senml message format handling in Magistrala

### DISCUSSION THREAD

**Discussion by ramu538** [absmach/magistrala#2341] [Q&A]
I am exploring Magistrala platform and trying to send SenML message using the curl command as shown below:
curl -s -S -i --cacert docker/ssl/certs/ca.crt -X POST -H "Content-Type: application/senml+json" -H "Authorization: Thing 1466973d-41f8-45d6-8d13-0cdd3b365ac3" https://localhost/http/channels/60d1061a-b9bf-403d-bb34-8843fbfdb3e4/messages -d '[{"bn":"some-base-name:","bt":1.276020076001e+09, "bu":"A","bver":5, "n":"voltage","u":"V","v":120.1}, {"n":"current","t":-5,"v":1.2}, {"n":"current","t":-4,"v":1.3}]' 

Following Successful Logs which are expected observed in timescale writer logs:
Processing SenML Record: {Channel:60d1061a-b9bf-403d-bb34-8843fbfdb3e4 Subtopic: Publisher:01f955c7-9bd9-44b7-ae31-f56cd4abbc58 Protocol:http Name:some-base-name:voltage Unit:V Time:1.276021076001e+18 UpdateTime:0 Value:0xc00031aa30 StringValue:<nil> DataValue:<nil> BoolValue:<nil> Sum:<nil>}
{"time":"2024-07-15T10:15:45.667512452Z","level":"INFO","msg":"Blocking consumer consumed messages successfully","duration":"222.563065ms"}

However, for the same Senml message following error observed using magistrala-smtp-notifier logs:
{"time":"2024-07-15T07:37:39.63840652Z","level":"WARN","msg":"Blocking consumer failed to consume messages successfully","duration":"30.648µs","error":{"error":"","message":"failed to convert to Magistrala message"}} {"time":"2024-07-15T07:37:39.638446225Z","level":"WARN","msg":"Blocking consumer failed to consume messages with error: ","!BADKEY":{"error":"","message":"failed to convert to Magistrala message"}}  

Able to receive and observe the message in timescale database which is being sent by curl command, however due to the above error message this message is not observed in subscriptions database.

Can you please let me know what might be the issue in the SMTP notifier for this message sent using curl command.

**Comment by dborovcanin** [comment]
Hello @ramu538, please create a bug report for these types of questions with details about the Magistrala version, environment, and tools you are using and format the code to make it more readable.
